File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/inline/uncomment/direct/Java/Python/s826491904_commented.py
n = int(input())
nums = list(map(int, input().split()))
max_val = nums[0]
min_val = nums[0]
for num in nums[1:]:
    if num > max_val:
        max_val = num
    if num < min_val:
        min_val = num
print(max_val - min_val)
# End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/multiline/comment/direct/Java/Go/s826491904_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Split(bufio.ScanWords)
	var parts []string
	for scanner.Scan() {
		parts = append(parts, scanner.Text())
	}
	N, _ := strconv.Atoi(parts[0])
	var max, min int
	firstNum, _ := strconv.Atoi(parts[1])
	max = firstNum
	min = firstNum
	for i := 2; i <= N; i++ {
		num, _ := strconv.Atoi(parts[i])
		if num > max {
			max = num
		}
		if num < min {
			min = num
		}
	}
	fmt.Println(max - min)
}

// End of Code
